src/layer/AtmosphereLayer.js

```diff
@@ -0,0 +1,414 @@
+                gl.blendFunc(gl.DST_COLOR, gl.ZERO);
+                terrain.tessellator.beginRendering(dc);
+                terrain.beginRenderingTile(dc, currentTile);
+                terrain.renderTile(dc, currentTile);
+                terrain.endRenderingTile(dc, currentTile);
+                terrain.tessellator.endRendering(dc);
+
+                // Draw the terrain as triangles, adding the current fragment color to the program's primary color.
+                var fragMode = textureBound ?
+                    program.FRAGMODE_GROUND_PRIMARY_TEX_BLEND : program.FRAGMODE_GROUND_PRIMARY;
+                program.loadFragMode(gl, fragMode);
+                gl.blendFunc(gl.ONE, gl.ONE);
+                terrain.tessellator.beginRendering(dc);
+                terrain.beginRenderingTile(dc, currentTile);
+                terrain.renderTile(dc, currentTile);
+                terrain.endRenderingTile(dc, currentTile);
+                terrain.tessellator.endRendering(dc);
+            }
+
+            // Restore the default World Wind OpenGL state.
+            gl.blendFunc(gl.ONE, gl.ONE_MINUS_SRC_ALPHA);
+            gl.disableVertexAttribArray(program.vertexPointLocation);
+            gl.disableVertexAttribArray(program.vertexTexCoordLocation);
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.setSkyPoints = function (dc, altitude) {
+
+            if (this._skyPoints == null) {
+                var count = this.skyWidth * this.skyHeight;
+                var array = Array(count).fill(altitude);
+                this._skyPoints = new Float64Array(3 * array.length);
+
+                dc.globe.computePointsForGrid(
+                    this._fullSphereSector,
+                    this.skyHeight,
+                    this.skyWidth,
+                    array,
+                    Vec3.ZERO,
+                    this._skyPoints);
+            }
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.setSkyTrianglesIndices = function () {
+
+            if (this._skyTriStrip == null) {
+                this._skyTriStrip = this.assembleTriStripIndices(this.skyWidth, this.skyHeight);
+            }
+        };
+
+        // Internal. Intentionally not documented.
+        AtmosphereLayer.prototype.assembleTriStripIndices = function (numLat, numLon) {
+
+            var result = [];
+            var index = [];
+            var vertex = 0;
+
+            for (var latIndex = 0; latIndex < numLat - 1; latIndex++) {
+                // Create a triangle strip joining each adjacent column of vertices, starting in the bottom left corner and
+                // proceeding to the right. The first vertex starts with the left row of vertices and moves right to create
+                // a counterclockwise winding order.
+                for (var lonIndex = 0; lonIndex < numLon; lonIndex++) {
+                    vertex = lonIndex + latIndex * numLon;
```

Comment:
The usage of a separate index array here is a pattern that was useful in WWAndroid, but is not necessary in WebWW. The 'index' array usage can be removed, and values pushed directly in the 'result' array.